src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,20 +2,109 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
 
-import static org.junit.Assert.assertTrue;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
 
-public class ArrayListProductDaoTest
-{
-    private ProductDao productDao;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ArrayListProductDaoTest {
+
+    @Mock
+    private Product product1;
+
+    @Mock
+    private Product product2;
+
+    @Mock
+    private Product productToSave;
+
+    @Spy
+    private ArrayList<Product> products;
+
+    @InjectMocks
+    private ArrayListProductDao productDao = ArrayListProductDao.getInstance();
 
     @Before
     public void setup() {
-        productDao = new ArrayListProductDao();
+        products.addAll(Arrays.asList(product1, product2));
+        when(product1.getId()).thenReturn(1L);
+        when(product1.getPrice()).thenReturn(new BigDecimal(200));
+        when(product1.getStock()).thenReturn(23);
+
+        when(product2.getId()).thenReturn(2L);
+        when(product2.getPrice()).thenReturn(new BigDecimal(300));
+        when(product2.getStock()).thenReturn(42);
+    }
+
+    @Test
+    public void testGetProduct() {
+        assertNotNull(productDao.getProduct(1L));
+    }
+
+    @Test(expected = ProductNotFoundException.class)
+    public void testGetProductException() {
+        productDao.getProduct(3L);
+    }
+
+    @Test
+    public void testFindProducts() {
+        assertEquals(2, productDao.findProducts().size());
     }
 
     @Test
     public void testFindProductsNoResults() {
+        products.clear();
+        assertEquals(0, productDao.findProducts().size());
+    }
+
+    @Test
+    public void testFindNullPriceProducts() {
+        when(product1.getPrice()).thenReturn(null);
+        assertEquals(1, productDao.findProducts().size());
+
+        when(product2.getPrice()).thenReturn(null);
```

Comment:
One tested object's method invocation for one test is enough, the other one is odd